pkg/restore/restore.go

```diff
@@ -696,6 +797,8 @@ func (ctx *context) restoreResource(resource, namespace, resourcePath string) (a
 				go func() {
 					defer ctx.resourceWaitGroup.Done()
 
+					// TODO(nrb): This is currently conflicting with the namespace termination waits. If the namespace becomes
+					// ready to restore, everything in the NS seems to be restored but the PV never gets created and the PVC is "lost"
```

Comment:
Thoughts on handling this? I'm going to keep testing and see if a new PV is made then deleted during the termination wait, but that may be a dead end.

Reply:
I don't really see why this would affect PVs on first glance - let me see if I can test it and figure out what's going on

Reply:
I believe it's cause of the PV timeout being 1 minute, where I was using 2-5 minute namespace timeouts. I believe the PV restore attempt would timeout before the namespace termination did, and thus abandon doing a restore. I did try using the same `ctx.namespaceTimeout` value for both, but I didn't see any improvement.

I *think* upping the timeout on PV restoration would cause an issue with mapping the PVC, but I'm not positive.

Reply:
OK, here's what I think is happening:

1. namespace `foo` has a PVC claiming PV `bar`
2. namespace `foo` is deleted and moves to `Terminating` state
3. user executes `ark restore create` which starts waiting on `foo` to be deleted
4. `foo` finishes deleting, at which point the PVC claiming PV `bar` is gone, and PV `bar` is released
5. Ark immediately restores resources in NS `foo`, then moves on to trying to restore the PV
6. The dynamic provisioner that deletes released PVs hasn't yet deleted the existing PV
7. Ark sees that it's trying to restore a resource that already exists, and can't, so it moves on
8. The dynamic provisioner deletes the PV
9. End result: Ark restores the PVC in namespace `foo`, but the PV never gets restored

Reply:
We'd need to also wait on PVs in the `Released` state with a reclaim policy of `Delete` to disappear before trying to restore them if we want to handle this correctly

Reply:
So you're saying this would be a change to the `waitForReady` function? Such as how we determine the value of `isPVReady`?

Reply:
I'm saying we'd need a `pvWaiter` that does basically the same thing as the `namespaceWaiter`, but for pvs

Reply:
Thanks!

Reply:
Can remove this comment now